bonsai/bonsai.go

```diff
@@ -0,0 +1,145 @@
+	// AssetName is the name of the Sensu asset
+	AssetName string `json:"asset_name,omitempty"`
+	// CurrentVersion is the version of the Sensu asset currently installed
+	CurrentVersion string `json:"current_version,omitempty"`
+	// LatestVersion is the latest version of the asset in Bonsai
+	LatestVersion string `json:"latest_version,omitempty"`
+}
+
+// NewBaseAsset creates a new BaseAsset
+func NewBaseAsset(name string) (*BaseAsset, error) {
+	bAsset := BaseAsset{}
+
+	versionSplit := strings.Split(name, ":")
+	if len(versionSplit) > 2 {
+		return nil, errors.New("only one colon is permitted in the asset name")
+	}
+	if len(versionSplit) == 2 {
+		bAsset.Version = versionSplit[1]
+	}
+	nameWithoutVersion := versionSplit[0]
+
+	nameSplit := strings.Split(nameWithoutVersion, "/")
+	if len(nameSplit) != 2 {
+		return nil, errors.New("asset name must be specified in the format \"namespace/asset\" (e.g. mynamespace/cpu-check)")
+	}
+	bAsset.Namespace = nameSplit[0]
+	bAsset.Name = nameSplit[1]
+
+	return &bAsset, nil
+}
+
+// HasVersion will return whether or not a Bonsai asset contains a specific version.
+func (b *Asset) HasVersion(version *goversion.Version) bool {
+	for _, validVersion := range b.ValidVersions() {
+		if validVersion.Equal(version) {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidVersions will return a list of unsorted semver-compliant versions.
+// Any versions that cannot be parsed as semver-compliant will be disregarded.
+func (b *Asset) ValidVersions() []*goversion.Version {
+	var versions []*goversion.Version
+
+	for _, bVersion := range b.Versions {
+		version, err := goversion.NewVersion(bVersion.Version)
+		if err == nil {
+			versions = append(versions, version)
+		}
+	}
+	return versions
+}
+
+// LatestVersion will return the highest semver-compliant version.
+func (b *Asset) LatestVersion() *goversion.Version {
+	versions := b.ValidVersions()
+	sort.Sort(goversion.Collection(versions))
+	return versions[len(versions)-1]
+}
+
+// GetObjectMeta is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) GetObjectMeta() corev2.ObjectMeta {
+	return corev2.ObjectMeta{}
+}
+
+// SetObjectMeta is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) SetObjectMeta(meta corev2.ObjectMeta) {
+	// no-op
+}
+
+// SetNamespace is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) SetNamespace(namespace string) {
+	// no-op
+}
+
+// StorePrefix is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) StorePrefix() string {
+	return ""
+}
+
+// RBACName is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) RBACName() string {
+	return ""
+}
+
+// URIPath is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) URIPath() string {
+	return ""
+}
+
+// Validate is just a no-op to satisfy the Resource interface
+func (o *OutdatedAsset) Validate() error {
+	return nil
+}
```

Comment:
Can be removed

Reply:
I think `OutdatedAsset` still needs to fulfill the requirements of a `Resource` so that it can utilize some CLI helpers:
- https://github.com/sensu/sensu-go/blob/refactor/bonsai/cli/commands/helpers/print.go#L20-L28
- https://github.com/sensu/sensu-go/blob/refactor/bonsai/cli/commands/asset/outdated.go#L109-L115

Happy to create a specific function if we'd rather not have `OutdatedAsset` as a `Resource`.